### PR TITLE
[9.2.x] Support upstream mptcp

### PR DESCRIPTION
**Pull request by serrislew**
Same as #10701 but for 9.2.x branch

### CODE REVIEW

**Comment by serrislew** [comment]
[approve ci debian]

**Comment by serrislew** [comment]
[approve ci clang-format]

**Comment by serrislew** [comment]
[approve ci]

**Comment by bneradt** [comment]
[approve ci]

File: iocore/net/Connection.cc [hunk 1]
```diff
@@ -229,7 +229,7 @@ Server::setup_fd_for_listen(bool non_blocking, const NetProcessor::AcceptOptions
 #endif
   }
 
-  if ((opt.sockopt_flags & NetVCOptions::SOCK_OPT_NO_DELAY) &&
+  if ((opt.sockopt_flags & NetVCOptions::SOCK_OPT_NO_DELAY) && !opt.f_mptcp &&
       safe_setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, SOCKOPT_ON, sizeof(int)) < 0) {
     goto Lerror;
   }
```
Comment (bneradt) [inline iocore/net/Connection.cc:235]: Can you help me understand this change? I assume TCP_NODELAY is not applicable to MPTCP?

File: iocore/net/Connection.cc [hunk 2]
```diff
@@ -390,6 +385,7 @@ Server::listen(bool non_blocking, const NetProcessor::AcceptOptions &opt)
     fd = NO_FD;
   }
 
-  Error("Could not bind or listen to port %d (error: %d)", ats_ip_port_host_order(&addr), res);
+  Fatal("Could not bind or listen to port %d, mptcp enabled: %d (error: %d) %s %d", ats_ip_port_host_order(&addr),
+        prot == IPPROTO_MPTCP, errno, strerror(errno), res);
```
Comment (bneradt) [inline iocore/net/Connection.cc:389]: This makes sense. If you configured to listen on a port and can't, probably best to Fatal.

**Review by bneradt — APPROVED** [review]
Looks fine. I have a question, but the change is intentional enough that I trust you're doing the right thing with the socket options.

File: iocore/net/Connection.cc (hunk 1 above)
Comment (serrislew) [reply iocore/net/Connection.cc:235]: It seems like now that we are using MPTCP as a protocol and not a socket option, the socket options for IPPROTO_TCP doesn't apply for sockets that are not defined for IPPROTO_TCP

File: iocore/net/Connection.cc (hunk 2 above)
Comment (serrislew) [reply iocore/net/Connection.cc:389]: Oh this is Fatal on master but maybe it should be consistent with 9.2.x. Thoughts?

Comment (bneradt) [reply iocore/net/Connection.cc:389]: Ah, gotcha. Yeah, probably fine to pull it in as fatal for 9.2.x too while we're at it.

**Comment by zwoop** [comment]
@serrislew Does this remove the support for the old MPTCP stack? That's fine (and expected) on master, but we wouldn't want that on 9.2.x branch.

**Pull request closed without merging**